sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1134,28 +1119,44 @@ _pi_queue::_pi_queue(std::vector<ze_command_queue_handle_t> &ComputeQueues,
   CopyCommandBatch.QueueBatchSize = ZeCommandListBatchCopyConfig.startSize();
 }
 
+static pi_result cleanup(pi_event);
```

Comment:
Why did you change this to be a free function instead of being a member of `_pi_event`?

Reply:
I now see your comment in the description:

> Make cleanup() and resetCommandLists() functions free. The resetCommandLists() function performs release of events in command lists and we release the queue if event's refcount reaches 0, so it is possible that queue can be destructed inside this function. That's why it should be free function and not a method of the queue. Same for cleanup().

What's wrong with `cleanup()` being a PI event's member function? How is this solved with it being a free function?

Reply:
Isn't it the same thing? Essentially an "event" should be considered dead after piEventRelease.
That's actually true for every "release" interface that it deletes the object if last reference to it is released.
So this release should be the last thing you do with an object.

Reply:
Just to clarify, this is not so important change and it is a matter of taste. It is highly recommended to avoid "delete this" that's why I've made this change because I was trying to improve things.  But it is ok to do "delete this" as long as we are careful: https://isocpp.org/wiki/faq/freestore-mgmt#delete-this

If you insist on having this as the method of pi_event I can revert this change.

Reply:
No, I am fine with free-standing. but please rename it then